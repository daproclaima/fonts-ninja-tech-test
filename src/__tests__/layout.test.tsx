import {cleanup, render} from "@testing-library/react";
import {beforeEach, describe, expect, test} from "vitest";
import Layout from "../app/layout";

describe("Feature: Layout", () => {
  beforeEach(() => {
    cleanup();
  });

  describe("Given the Layout component is rendered", () => {
    const renderLayout = () => render(<Layout>children</Layout>);

    describe("When is on server", () => {
      test("Then the content is present", () => {
        const { container } = renderLayout()

        expect(container).toMatchSnapshot()
      });
    });
  });
});
