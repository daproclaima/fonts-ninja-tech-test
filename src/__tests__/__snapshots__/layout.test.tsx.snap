// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Feature: Layout > Given the Layout component is rendered > When is on server > Then the content is present 1`] = `
<div>
  <header
    class="px-14 pt-12 flex flex-row justify-between"
  >
    <a
      aria-label="return to home page"
      href="/"
    >
      <img
        alt="fonts ninja logo"
        class="dark:invert"
        data-nimg="1"
        decoding="async"
        height="26"
        src="/fonts-ninja.svg"
        style="color: transparent;"
        width="24"
      />
    </a>
    <button
      class="w-[150px] py-4 px-2 bg-coral rounded-2xl first-letter:uppercase"
      type="button"
    >
      switch theme
    </button>
  </header>
  children
</div>
`;
